openmc/stats/multivariate.py

```diff
@@ -756,6 +758,125 @@ def from_xml_element(cls, elem, meshes):
         return cls(meshes[mesh_id], strengths, volume_normalized)
 
 
+class PointCloud(Spatial):
+    """Spatial distribution from a point cloud.
+
+    This distribution specifies a discrete list of points, 
+    each with different relative probability.
+
+    .. versionadded:: 0.15.x
+
+    Parameters
+    ----------
+    positions : iterable of 3-tuples
+        The points in space to be sampled
+    strengths : iterable of float, optional
+        An iterable of values that represents the relative probabilty of each point.
+
+    Attributes
+    ----------
+    psoitions: numpy.ndarray (3xN)
+        The points in space to be sampled
+    strengths : numpy.ndarray or None
+        An array of relative probabilities for each mesh point
+    """
+
+    def __init__(self, positions, strengths=None):
+        self.positions = positions
+        self.strengths = strengths
+
+    @property
+    def positions(self):
+        return self._positions
+
+    @positions.setter
+    def positions(self, given_positions):
+        if given_positions is None:
+            raise ValueError('No positions were provided')
+        cv.check_iterable_type('position list passed in', given_positions, Real, 2, 2)
+
+        if isinstance(given_positions, list):
+            cv.check_length('first position entry', given_positions[0], 3, 3)
+            self._positions = np.asarray(given_positions)
+        elif isinstance(given_positions, np.ndarray):
+            self._positions = given_positions
```

Comment:
Another place to seek OpenMC design best practice about being tolerant to lots of data forms for this data

Reply:
The trend seems to be storing NumPy arrays for the internal attributes values when in doubt. I'll let @paulromano chime in here though in case his perspective differs.

Reply:
But is that imposed as a requirement? or are we tolerant of other forms?

Reply:
Tolerant of other forms I'd say.